exchanges/bybit/bybit.go

```diff
@@ -8,740 +8,2348 @@ import (
+	params, err := fillCategoryAndSymbol(category, symbol)
+	if err != nil {
+		return nil, err
+	}
+	if category == cOption && baseCoin == "" {
+		return nil, errBaseNotSet
+	}
+	if baseCoin != "" {
+		params.Set("baseCoin", baseCoin)
+	}
+	if optionType != "" {
+		params.Set("optionType", optionType)
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp *TradingHistory
+	return resp, by.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues("market/recent-trade", params), defaultEPL, &resp)
+}
+
+// GetOpenInterest retrieves open interest of each symbol.
+func (by *Bybit) GetOpenInterest(ctx context.Context, category, symbol, intervalTime string, startTime, endTime time.Time, limit int64, cursor string) (*OpenInterest, error) {
+	if category == "" {
+		return nil, errCategoryNotSet
+	} else if category != cLinear && category != cInverse {
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, category)
+	}
+	params := url.Values{}
+	if symbol == "" {
+		return nil, errSymbolMissing
+	}
+	params.Set("symbol", symbol)
+	params.Set("category", category)
+	if intervalTime != "" {
+		params.Set("intervalTime", intervalTime)
+	}
+	if !startTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if cursor != "" {
+		params.Set("cursor", cursor)
+	}
+	var resp *OpenInterest
+	return resp, by.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues("market/open-interest", params), defaultEPL, &resp)
+}
+
+// GetHistoricalValatility retrieves option historical volatility.
+// The data is hourly.
+// If both 'startTime' and 'endTime' are not specified, it will return the most recent 1 hours worth of data.
+// 'startTime' and 'endTime' are a pair of params. Either both are passed or they are not passed at all.
+// This endpoint can query the last 2 years worth of data, but make sure [endTime - startTime] <= 30 days.
+func (by *Bybit) GetHistoricalValatility(ctx context.Context, category, baseCoin string, period int64, startTime, endTime time.Time) ([]HistoricVolatility, error) {
+	if category == "" {
+		return nil, errCategoryNotSet
+	} else if category != cOption {
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, category)
+	}
+	params := url.Values{}
+	params.Set("category", category)
+	if baseCoin != "" {
+		params.Set("baseCoin", baseCoin)
+	}
+	if period > 0 {
+		params.Set("period", strconv.FormatInt(period, 10))
+	}
+	var err error
+	if err = common.StartEndTimeCheck(startTime, endTime); !(startTime.IsZero() && endTime.IsZero()) && err != nil {
+		return nil, err
+	} else if !startTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
```

Comment:
I don't think this fully matches the criteria. I think this statement matches it more, but happy for feedback
```go
if !startTime.IsZero() || !endTime.IsZero() {
		err = common.StartEndTimeCheck(startTime, endTime)
		if err != nil {
			return nil, err
		}
		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
	}
```